src/script/interpreter.cpp

```diff
@@ -274,6 +274,35 @@ int FindAndDelete(CScript& script, const CScript& b)
     return nFound;
 }
 
+/** Helper for OP_CHECKSIG and OP_CHECKSIGVERIFY
+ *
+ * A return value of false means the script fails entirely. When true is returned, the
+ * fSuccess variable indicates whether the signature check itself succeeded.
+ */
+static bool EvalChecksig(const valtype& vchSig, const valtype& vchPubKey, CScript::const_iterator pbegincodehash, CScript::const_iterator pend, unsigned int flags, const BaseSignatureChecker& checker, SigVersion sigversion, ScriptError* serror, bool& fSuccess)
+{
+    // Subset of script starting at the most recent codeseparator
+    CScript scriptCode(pbegincodehash, pend);
+
+    // Drop the signature in pre-segwit scripts but not segwit scripts
```

Comment:
18422:
```suggestion
    // Drop the signature, since there's no way for a signature to sign itself
```

Reply:
still not fixed, see https://github.com/dashpay/dash/pull/5143/commits/c88389400d9677454573f061311ab4a0ecea2f5b#diff-a0337ffd7259e8c7c9a7786d6dbd420c80abfa1afdb34ebae3261109d9ae3c19L984

Reply:
fixed now